File: ios-app/Modules/Sources/Domain/Models/API/ActivityInput.swift
import Foundation

public struct ActivityInput: Equatable, Sendable {
    public var title: String
    public var agenda: String?
    public var questions: [EventInput.QuestionInput]
    public var runMode: RunMode
    public var invitedEmails: [String]
    public var sendEmails: Bool

    public init(
        title: String,
        agenda: String? = nil,
        questions: [EventInput.QuestionInput] = [],
        runMode: RunMode = .manual,
        invitedEmails: [String] = [],
        sendEmails: Bool = false
    ) {
        self.title = title
        self.agenda = agenda
        self.questions = questions
        self.runMode = runMode
        self.invitedEmails = invitedEmails
        self.sendEmails = sendEmails
    }
}

public extension ActivityInput {
    init(_ eventInput: EventInput) {
        self.init(
            title: eventInput.title,
            agenda: eventInput.agenda,
            questions: eventInput.questions,
            runMode: .manual,
            invitedEmails: [],
            sendEmails: false
        )
    }

    enum RunMode: String, CaseIterable, Hashable, Sendable {
        case manual
        case automatic
    }
}

File: backend/apps/api/src/main/kotlin/dk/example/feedback/dto/ActivityInput.kt
package dk.example.feedback.dto

import dk.example.feedback.model.enumerations.ActivityRunMode
import dk.example.feedback.payloads.QuestionInput
import io.swagger.v3.oas.annotations.media.Schema

@Schema(description = "Manager request payload for creating or updating an activity.")
data class ActivityInput(
    @field:Schema(description = "Activity title shown in manager and participant views.")
    val title: String,
    val agenda: String?,
    @field:Schema(description = "Ordered list of activity questions.")
    val questions: List<QuestionInput>,
    @field:Schema(description = "Run mode controlling event participation behavior.")
    val runMode: ActivityRunMode,
    val invitedEmails: List<String> = emptyList(),
    val sendEmails: Boolean,
)
